vulnerable_compilable.cpp

```
@@ -0,0 +1,110 @@
+
+        // 多執行緒競爭更新全域變數（未使用同步機制）
+        thread t1(&VulnerableServer::incrementCounter, this);
+        thread t2(&VulnerableServer::incrementCounter, this);
+        t1.join();
+        t2.join();
+        cout << "Global Counter: " << globalCounter << endl;
+
+        // 透過 new 配置記憶體後，delete 後仍使用（野指標問題）
+        int *dangling = new int(42);
+        delete dangling;
+        cout << "Dangling pointer value: " << *dangling << endl;
```

Cppcheck (2.10-2)

[error] Dereferencing 'dangling' after it is deallocated / released

(deallocuse)